### PR TITLE
Issue 1171: Dropping privileges from authorization configuration

### LINKED ISSUE #1171: Drop privileges from authorization configuration
# Task Description

We should drop the `privileges` field from the following classes:

- `org.carlspring.strongbox.authorization.dto.AuthorizationConfigDto`
- `org.carlspring.strongbox.authorization.domain.AuthorizationConfig`

It will also affect `strongbox-authorization.yaml`/`strongbox-authorization.xml`.

@carlspring I don't see any benefit from keeping this field. How do you want to use these privileges ? Currently, we are not able to use them in any form and we should drop them because it may cause confusions.

If you agree, please take the `on hold` label off from this task. 

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by ogryniuk**
Fixes #1171 

1. Removed privileges fields from the classes [AuthorizationConfigDto](https://github.com/strongbox/strongbox/blob/8960501406eaad51958a00427ecb2dd7a4329605/strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/authorization/dto/AuthorizationConfigDto.java) and [AuthorizationConfig](https://github.com/strongbox/strongbox/blob/c12bfbf7b5f2611c53954b1429b1128250b6c3ad/strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/authorization/domain/AuthorizationConfig.java)
2. Removed [authorizationConfiguration.privileges ](https://github.com/ogryniuk/strongbox/commit/c8c157de1b2082914c40dcf1353ec2a8764ad345#diff-045acd1c1929e3ce3b4875dcb519d296L16)
3. Improved the syntax of LdapAuthenticationProviderTest ([deleted unnecessary empty spaces](https://github.com/strongbox/strongbox/blob/6462384966d1230bbd2e4cc8ee9db940213f9c1b/strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationProviderTest.java#L109))

### CODE REVIEW

File: strongbox-security/strongbox-user-management/src/main/resources/etc/conf/strongbox-authorization.yaml [hunk 1]
```diff
@@ -2,18 +2,7 @@ authorizationConfiguration:
   roles:
     - name: ANONYMOUS_ROLE
       description: Common anonymous user role
-      privileges:
```
Comment (sbespalov) [inline strongbox-security/strongbox-user-management/src/main/resources/etc/conf/strongbox-authorization.yaml]: @ogryniuk you shouldn't drop this `privileges`, only root `privileges` attribute should be dropped

File: strongbox-security/strongbox-user-management/src/main/resources/etc/conf/strongbox-authorization.yaml [hunk 2]
```diff
@@ -2,18 +2,7 @@ authorizationConfiguration:
...
-        - ARTIFACTS_RESOLVE
-        - SEARCH_ARTIFACTS
     - name: USER_ROLE
       description: Common user role
-      privileges:
```
Comment (sbespalov) [inline strongbox-security/strongbox-user-management/src/main/resources/etc/conf/strongbox-authorization.yaml]: same as above

File: strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationProviderTest.java [hunk 3]
```diff
@@ -80,8 +80,7 @@ public void embeddedLdapServerRegistersExpectedAuthenticationProvider()
         assertThat(ldapUser.getAuthorities(),
                    CoreMatchers.hasItems(CoreMatchers.equalTo((GrantedAuthority) Privileges.ADMIN_CREATE_REPO),
                                          CoreMatchers.equalTo((GrantedAuthority) Privileges.ADMIN_DELETE_REPO),
-                                         CoreMatchers.equalTo((GrantedAuthority) Privileges.ADMIN_CREATE_REPO),
-                                         CoreMatchers.equalTo(new SimpleGrantedAuthority("VIEW_USER"))));
```
Comment (sbespalov) [inline strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationProviderTest.java]: @ogryniuk the code changes shouldn't affect on this test

**Comment by ogryniuk** [comment]
@sbespalov 

Sure, I did this already. Thanks for your feedback :)

**Comment by sbespalov** [comment]
@ogryniuk there seems test failures in Jenkins, can you please have a look as well?

**Comment by ogryniuk** [comment]
@sbespalov 

Without changes to strongbox-authorization.yaml, the build fails because of 

> Unrecognized field "privileges"

[org.carlspring.strongbox.users.service.AuthorizationConfigServiceTest.txt](https://github.com/strongbox/strongbox/files/3289473/org.carlspring.strongbox.users.service.AuthorizationConfigServiceTest.txt)
[org.carlspring.strongbox.users.service.UserServiceTest.txt](https://github.com/strongbox/strongbox/files/3289474/org.carlspring.strongbox.users.service.UserServiceTest.txt)

**Comment by sbespalov** [comment]
> Without changes to strongbox-authorization.yaml, the build fails because of

yeah, but you should drop it from `yaml` too

**Comment by ogryniuk** [comment]
@sbespalov 

Realised it a second before reading your message. My bad. Thanks

**Comment by ogryniuk** [comment]
@sbespalov 

It is done. 

Cannot say about GitHub Jenkins builds but on my local machine everything gets built fine

**Comment by sbespalov** [comment]
@steve-todorov seems that there is Jenkins issues within this PR build.

**Comment by steve-todorov** [comment]
@sbespalov / @ogryniuk thanks for reporting! Tests are passing now :)

**Comment by carlspring** [comment]
Thanks!

**Pull request merged**